src/driver/kvm.rs

```diff
@@ -84,13 +87,88 @@ impl Introspectable for Kvm {
             r15: regs.r15,
             rip: regs.rip,
             rflags: regs.rflags,
-            cr0: sregs.cr0,
-            cr3: sregs.cr3,
-            cr4: sregs.cr4,
-            fs_base: sregs.fs.base,
+            cr0: _sregs.cr0,
+            cr2: _sregs.cr2,
+            cr3: _sregs.cr3,
+            cr4: _sregs.cr4,
+            sysenter_cs: _msrs.entries[0].data,
+            sysenter_esp: _msrs.entries[1].data,
+            sysenter_eip: _msrs.entries[2].data,
+            msr_efer: _msrs.entries[3].data,
+            msr_star: _msrs.entries[4].data,
+            msr_lstar: _msrs.entries[5].data,
+            efer: _sregs.efer,
+            apic_base: _sregs.apic_base,
+            cs: segment_reg {
+                base: _sregs.cs.base,
+                limit: _sregs.cs.limit,
+                selector: _sregs.cs.selector,
+            },
+            ds: segment_reg {
+                base: _sregs.ds.base,
+                limit: _sregs.ds.limit,
+                selector: _sregs.ds.selector,
+            },
+            es: segment_reg {
+                base: _sregs.es.base,
+                limit: _sregs.es.limit,
+                selector: _sregs.es.selector,
+            },
+            fs: segment_reg {
+                base: _sregs.fs.base,
+                limit: _sregs.fs.limit,
+                selector: _sregs.fs.selector,
+            },
+            gs: segment_reg {
+                base: _sregs.gs.base,
+                limit: _sregs.gs.limit,
+                selector: _sregs.gs.selector,
+            },
+            ss: segment_reg {
+                base: _sregs.ss.base,
+                limit: _sregs.ss.limit,
+                selector: _sregs.ss.selector,
+            },
+            tr: segment_reg {
+                base: _sregs.tr.base,
+                limit: _sregs.tr.limit,
+                selector: _sregs.tr.selector,
+            },
+            ldt: segment_reg {
+                base: _sregs.ldt.base,
+                limit: _sregs.ldt.limit,
+                selector: _sregs.ldt.selector,
+            },
         }))
     }
 
+    fn write_registers(&self, vcpu: u16, value: u64, reg: Register) -> Result<(), Box<dyn Error>> {
+        let (mut regs, _sregs, _msrs) = self.kvmi.get_registers(vcpu)?;
+        match reg {
+            a if a == Register::RAX => regs.rax = value,
```

Comment:
`Register::RAX => regs.rax = value,` would be better

Reply:
Also, an even cleaner way to perform conversion would be to use the from or into trait. Have a look at https://doc.rust-lang.org/stable/rust-by-example/conversion/from_into.html